python_modules/libraries/dagster-aws/dagster_aws/pipes.py

```diff
@@ -102,3 +116,114 @@ def no_messages_debug_text(self) -> str:
             " PipesS3MessageWriter to be explicitly passed to open_dagster_pipes in the external"
             " process."
         )
+
+
+@experimental
+class PipesLambdaLogsMessageReader(PipesMessageReader):
```

Comment:
Can you document it's capabilities. In particular the limitation on log length and what the product experience would be if you use this.

Reply:
Is it possible to implement this using the `PipesLogReader` API used for databricks? If I'm reading this correctly this doesn't support continuous polling of the logs.

Or does that not work/is otherwise inappropriate for lambda?

Reply:
The lambda sync invoke API gives you back the last 4k of logs directly in the response, so this initial version is exploring that. While it does have its limitations, its very simple and as long as the meaningful messages are towards the end of the computation (which i believe would be common) then it should work. 

The full logs are available via Cloud Watch so need to explore what the exact constraints of using those APIs to trail the logs are (likely via `PipesLogReader`).